btree/btree.c

```diff
@@ -2944,14 +3007,14 @@ static int64_t btree_put_tick(struct m0_sm_op *smop)
 				else
 					bop->bo_flags |= BOF_LOCKALL;
 			}
-			if (bop->bo_arbor->t_height <= tree->t_height) {
-				/* If height increased */
-				//return m0_sm_op_sub(&bop->bo_op, P_CLEANUP,
-				//                    P_INIT);
+			if (bop->bo_arbor->t_height != tree->t_height) {
+				/* If height decreased or increased */
 				lock_op_unlock(tree);
-				return P_CLEANUP;
+				return m0_sm_op_sub(&bop->bo_op, P_CLEANUP,
+					            P_INIT);
 			} else {
-				/* If height decreased */
+				/* If height same */
+				bop->bo_arbor->t_height = tree->t_height;
```

Comment:
If  height is same then no need of re-initializing t_height.

Reply:
Yes